tf_encrypted/keras/layers/normalization_test.py

```diff
@@ -0,0 +1,50 @@
+# pylint: disable=missing-docstring
+import unittest
+
+import numpy as np
+import tensorflow as tf
+
+import tf_encrypted as tfe
+from tf_encrypted.keras.testing_utils import agreement_test, layer_test
+
+np.random.seed(42)
+
+
+class TestBatchNormalization(unittest.TestCase):
+
+  def setUp(self):
+    tf.reset_default_graph()
+
+  def test_batchnorm_2d(self):
+    self._core_batchnorm([1, 4], axis=1)
+
+  def test_batchnorm_4d(self):
+    self._core_batchnorm([1, 1, 1, 3])
+
+  def test_batchnorm_channels_first(self):
+    self._core_batchnorm([1, 3, 1, 1], axis=1)
+
+  def test_batchnorm_no_scale(self):
+    self._core_batchnorm([1, 1, 1, 3], scale=False)
+
+  def test_batchnorm_no_center(self):
+    self._core_batchnorm([1, 1, 1, 3], center=False)
+
+
+  def _core_batchnorm(self, input_shape, **layer_kwargs):
+    base_kwargs = {
+        'fused': False
```

Comment:
Out of curiosity, what happen if fused = True?

Reply:
The tensorflow fused implementation is limited in where the channel can be. Can't remember if its first or last but the test will fail when it tries to compare the keras answer to our answer because it can't calculate the keras answer.

Reply:
Ok great, that makes sense.